### PR TITLE
Add support to regex lines starting with ~~~

### PR DESCRIPTION
**Pull request by phadej**
There are cases where ... is just not enough, e.g. when matching against
possible error messages of GHC

Motivated by https://github.com/haskell-servant/servant/issues/698 and
https://github.com/haskell-servant/servant/pull/699
Making a doctests for "should not type check", working across
GHC 7.8.4 - GHC 8.0.2 is practically impossible without regular
expression.

*EDIT* compiles also with GHC-7.0.4.

### CODE REVIEW

**Comment by sol** [comment]
@phadej Sorry for the late reply.  Before considering this, I would want to see a convincing example where this feature makes user documentation better.

I see how this could be useful for testing code.

***But:*** Contrary to popular believe, Doctest is all about testing examples in user documentation, not about testing code (that is make sure that examples in user documentation do actually work).

**Comment by phadej** [comment]
@sol, the `servant` example is documentation; we document which case should not compile. And with GHC-8.0 we want to show examples of `TypeError`s, but that is impossible if we want to verify examples with other GHC versions. I.e. 

```
~~~(Our custom error|Could not deduce)
```
would work for us in http://hackage.haskell.org/package/servant-0.10/docs/Servant-Utils-Links.html 

I admit that regex makes documentation hard to parse.

Alternative is to be able to provide alternative of lines. Essentially making "line regex", where
literal string is a literal character, `...` is a wildcard and *tbd* would make groups optional,  maybe something like:

```
>>> safeLink api bad_link
...
(((
...Our custom type error
|||
...Could not deduce
)))
```
